### PR TITLE
ECAM automatic switching

### PR DESCRIPTION
**Pull request by lukecologne**
**Summary of Changes**

* Adds dynamic switching of ECAM pages when no page is selected.

**Screenshots (if necessary)**

* Modeled after this graphby discord user Ares#3394:
grafik 

**Additional context**

* Still got an issue, when the engine mode selector is turned to crank or ignition start, not only does the ecam page open up, but the button gets illuminated as well. It's just visual though, and the page switches back to normal as expected.

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 1]
```diff
@@ -174,51 +180,80 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
...
-            }
+        var currentAPUMasterState = SimVar.GetSimVarValue("FUELSYSTEM VALVE SWITCH:8", "Bool");
 
+
+        //Determine displayed page when no button is selected
+        const prevPage = this.pageNameWhenUnselected;
+
+        const altitude = Simplane.getAltitude();
+        const isGearExtended = SimVar.GetSimVarValue("GEAR HANDLE POSITION", "Bool");
+        const currFlightPhase = SimVar.GetSimVarValue("RADIO HEIGHT", "feet") > 1500; //This simvar should be exactly what we need, but it doesn't work: SimVar.GetSimVarValue("L:AIRLINER_FLIGHT_PHASE", "number");
```
Comment (tyler58546) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:191]: What is the issue with the flight phase simvar?

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 2]
```diff
@@ -174,51 +180,80 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
...
+        const EngModeSel = SimVar.GetSimVarValue("L:XMLVAR_ENG_MODE_SEL", "number");
+        const spoilerOrFlapsDeployed = SimVar.GetSimVarValue("FLAPS HANDLE INDEX", "number") != 0 || SimVar.GetSimVarValue("SPOILERS HANDLE POSITION", "percent") != 0;
+
+        const crzCond = ((spoilerOrFlapsDeployed || ToPowerSet) && (currFlightPhase == true) && this.CrzCondTimer <= 0) || (currFlightPhase == true && !spoilerOrFlapsDeployed && !ToPowerSet);
+
+        if ((currFlightPhase == false) || (!spoilerOrFlapsDeployed && !ToPowerSet)) {
+            this.CrzCondTimer = 60;
+        } else if ((spoilerOrFlapsDeployed || ToPowerSet) && (currFlightPhase == true) && this.CrzCondTimer >= 0) {
+            this.CrzCondTimer -= _deltaTime/1000;
         }
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:206]: `== true` and `== false` are unnecessary.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js [hunk 3]
```diff
@@ -174,51 +180,80 @@ class A320_Neo_EICAS extends Airliners.BaseEICAS {
...
+            } else if (this.MainEngineStarterOffTimer >= 0) {
+                this.MainEngineStarterOffTimer -= _deltaTime/1000;
+            }
+            this.pageNameWhenUnselected = "ENG";
+        } else if (currentAPUMasterState && (APUPctRPM <= 95 || this.ApuAboveThresholdTimer >= 0)) {
+            // Show Apu when master sw. is on, and N% is lower than 95, or until 10s after N% over 95
+            if (this.ApuAboveThresholdTimer <= 0 && APUPctRPM <= 95) {
+                this.ApuAboveThresholdTimer = 10;
+            } else if (APUPctRPM >= 95) {
+                this.ApuAboveThresholdTimer -= _deltaTime/1000;
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:220]:
```suggestion
                this.ApuAboveThresholdTimer -= _deltaTime / 1000;
```

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js (hunk 1 above)
Comment (lukecologne) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/A320_Neo_EICAS.js:191]: Hm. Nothing it seems, maybe I made a mistake while testing it out. Will use this instead, but I only tested it with the current version. Should be functionally identical though

**Comment by wpine215** [comment]
I believe the reason behind the engine mode selector bug is that it calls the ECAM page switch with XML RPN code instead of a normal javascript function, so it bypasses your code completely. Not a game-breaking bug, so we can take a look at it alter on.

**Comment by lukecologne** [comment]
Yeah, as I said it's just visual, so no big deal. I think before i modified it, the page would disappear even if the mode selector was in IGN, but now it behaves as expected, so I'm guessing it doesn't bypass my code, but more or less does something in addition to us.

**Review by tyler58546 — APPROVED** [review]

**Pull request merged**